tooling/hcpctl/cmd/must-gather/cmd.go

```diff
@@ -0,0 +1,259 @@
+		Aliases: []string{"mg"},
+		Short:   "Azure Data Explorer must-gather operations",
+		GroupID: group,
+		Long: `must-gather provides data collection operations for Azure Data Explorer clusters.
+
+This command group includes subcommands for querying Azure Data Explorer instances
+and collecting diagnostic data for troubleshooting and analysis.`,
+		Example: `  hcpctl must-gather query --kusto-endpoint https://my-kusto-cluster.eastus.kusto.windows.net
+  hcpctl must-gather query --kusto-endpoint https://my-kusto-cluster.eastus.kusto.windows.net --output results.json`,
+		CompletionOptions: cobra.CompletionOptions{
+			HiddenDefaultCmd: true,
+		},
+	}
+
+	// Add query subcommand
+	queryCmd, err := newQueryCommand()
+	if err != nil {
+		return nil, err
+	}
+	cmd.AddCommand(queryCmd)
+
+	// Add show-builtin-queries subcommand
+	showBuiltinQueriesCmd, err := showBuiltinQueries()
+	if err != nil {
+		return nil, err
+	}
+	cmd.AddCommand(showBuiltinQueriesCmd)
+
+	return cmd, nil
+}
+
+func newQueryCommand() (*cobra.Command, error) {
+	opts := DefaultMustGatherOptions()
+
+	cmd := &cobra.Command{
+		Use:              "query",
+		Aliases:          []string{"q"},
+		Short:            "Execute default queries against Azure Data Explorer",
+		Long:             `Execute preconfigured queries against Azure Data Explorer clusters.`,
+		Args:             cobra.NoArgs,
+		SilenceUsage:     true,
+		SilenceErrors:    true,
+		TraverseChildren: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runQuery(cmd.Context(), opts)
+		},
+		CompletionOptions: cobra.CompletionOptions{
+			HiddenDefaultCmd: true,
+		},
+	}
+
+	if err := BindMustGatherOptions(opts, cmd); err != nil {
+		return nil, err
+	}
+
+	return cmd, nil
+}
+
+func showBuiltinQueries() (*cobra.Command, error) {
+	opts := DefaultMustGatherOptions()
+
+	cmd := &cobra.Command{
+		Use:     "show-builtin-queries",
+		Aliases: []string{"sbq"},
+		Short:   "Show builtin queries",
+		Long:    `Show builtin queries`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runShowBuiltinQueries(cmd.Context(), opts)
+		},
+		CompletionOptions: cobra.CompletionOptions{
+			HiddenDefaultCmd: true,
+		},
+	}
+
+	return cmd, nil
+}
+
+func runShowBuiltinQueries(_ context.Context, _ *RawMustGatherOptions) error {
+	queries := getServicesQueries([]string{"cid"}, "subscription", "rg")
+
+	for _, query := range queries {
+		fmt.Printf("Query %s:\n", query.Name)
+		fmt.Printf("--------------------------------\n")
+		fmt.Printf("%s\n", query.Query)
+		fmt.Printf("--------------------------------\n")
+	}
+
+	return nil
+}
+
+func runQuery(ctx context.Context, opts *RawMustGatherOptions) error {
+	// Validate options
+	validated, err := opts.Validate(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Complete options
+	completed, err := validated.Complete(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Create Kusto client
+	client, err := kusto.NewClient(opts.KustoEndpoint)
```

Comment:
Creating this client in the `Complete` and then using it in the `Run` will make unit testing `.Run` with a mock much easier.